### PR TITLE
chore: improve resiliency filtering aws subnets

### PR DESCRIPTION
**Pull request by esierra-stratio**

### CODE REVIEW

File: pkg/cluster/internal/create/actions/createworker/aws.go [hunk 1]
```diff
@@ -257,14 +257,16 @@ func filterPrivateSubnet(svc *ec2.EC2, subnetID *string) (string, error) {
...
 		for i := range associatedRouteTable.Routes {
-			if associatedRouteTable.Routes[i].DestinationCidrBlock != nil &&
-				associatedRouteTable.Routes[i].GatewayId != nil &&
-				*associatedRouteTable.Routes[i].DestinationCidrBlock == "0.0.0.0/0" &&
-				strings.Contains(*associatedRouteTable.Routes[i].GatewayId, "igw") {
-				isPublic = true
+			route := associatedRouteTable.Routes[i]
+
+			if route.DestinationCidrBlock != nil && route.GatewayId != nil {
+				if *route.DestinationCidrBlock == "0.0.0.0/0" && strings.Contains(*route.GatewayId, "igw") {
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/aws.go:263]: Why do we have nested ifs ?

File: pkg/cluster/internal/create/actions/createworker/aws.go [hunk 2]
```diff
@@ -288,14 +290,16 @@ func filterPublicSubnet(svc *ec2.EC2, subnetID *string) (string, error) {
...
```
Comment (stg-0) [inline pkg/cluster/internal/create/actions/createworker/aws.go:296]: Same as above, why do we have nested ifs ?

File: pkg/cmd/kind/create/cluster/validation/eksvalidator.go [hunk 3]
```diff
@@ -207,14 +207,16 @@ func filterPrivateSubnet(svc *ec2.EC2, subnetID *string) (string, error) {
...
 		for i := range associatedRouteTable.Routes {
-			if associatedRouteTable.Routes[i].DestinationCidrBlock != nil &&
-				associatedRouteTable.Routes[i].GatewayId != nil &&
-				*associatedRouteTable.Routes[i].DestinationCidrBlock == "0.0.0.0/0" &&
-				strings.Contains(*associatedRouteTable.Routes[i].GatewayId, "igw") {
-				isPublic = true
+			route := associatedRouteTable.Routes[i]
+
+			if route.DestinationCidrBlock != nil && route.GatewayId != nil {
+				if *route.DestinationCidrBlock == "0.0.0.0/0" && strings.Contains(*route.GatewayId, "igw") {
```
Comment (stg-0) [inline pkg/cmd/kind/create/cluster/validation/eksvalidator.go:213]: Same as above, why do we have nested ifs ?

File: pkg/cluster/internal/create/actions/createworker/aws.go (hunk 1 above)
Comment (esierra-stratio) [reply pkg/cluster/internal/create/actions/createworker/aws.go:263]: just for better understanding

**Review by stg-0 — APPROVED** [review]
lgtm

**Pull request merged**